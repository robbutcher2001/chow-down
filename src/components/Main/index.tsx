import React, { ReactNode } from 'react';

import styled from 'styled-components';
import { small } from '../../breakpoints';

import { RouterLink } from '../Clickable';

export interface CallToAction {
  text: string,
  link: string
}

interface MainProps {
  title?: string,
  cta?: CallToAction,
  children: ReactNode
};

const StyledZeroMarginedMain = styled.main`
  display: flex;
  justify-content: center;
  margin: 1rem 0 8rem 0;

  > div {
    max-width: 1100px;
    width: 100%; //can this be improved by using flexbox?

    > section:first-child {
      display: flex;
      flex-direction: row;
      flex-wrap: wrap;
      justify-content: space-between;
      align-items: center;
      margin: 0 1rem;

      h2 {
        margin: 1.5rem 0;
        font-size: ${props =>
          props.theme.typography.fontSize.xxlarge
        };
        color: ${props =>
          props.theme.colour.turquoise
        };

        ${small`
          margin: 0.5rem 2rem 0.5rem 0;
        `}
      }
    }
  }
`

const StyledMarginedMain = styled(StyledZeroMarginedMain)`
  margin: 1rem 1rem 8rem 1rem;

  > div > section:first-child {
    margin: unset;
  }
`

const Main = (props: MainProps) => (
  <div>
    <section>
      {props.title &&
        <h2>{props.title}</h2>
      }
      {props.cta &&
        <RouterLink to={props.cta.link} >
          {props.cta.text}
        </RouterLink>
      }
    </section>
    {props.children}
  </div>
);

export const ZeroMarginedMain = (props: MainProps) => (
  <StyledZeroMarginedMain>
    <Main {...props} />
  </StyledZeroMarginedMain>
);

export default (props: MainProps) => (
  <StyledMarginedMain>
    <Main {...props} />
  </StyledMarginedMain>
);