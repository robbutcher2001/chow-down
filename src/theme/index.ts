import { DefaultTheme } from 'styled-components';

import colour from './defs/_colour';

const theme: DefaultTheme = {
  colour
};

export default theme;