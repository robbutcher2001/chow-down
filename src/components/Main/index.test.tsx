import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { ThemeProvider } from 'styled-components';

import theme from '../../theme';
import Main from '.';

test('Main basic snapshot render', () => {
  const main = renderer.create(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <Main
          title='fake_title'
          cta={{
            text: 'fake_cta',
            link: 'fake_link'
          }}
        >
          <div>fake_child</div>
        </Main>
      </ThemeProvider>
    </Router>
  );

  expect(main.toJSON()).toMatchSnapshot();
});

test('Main correct tag content assertion', () => {
  const { getByText } = render(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <Main
          title='fake_title'
          cta={{
            text: 'fake_cta',
            link: 'fake_link'
          }}
        >
          <div>fake_child</div>
        </Main>
      </ThemeProvider>
    </Router>
  );

  expect(getByText(/fake_title/).textContent).toEqual('fake_title');
  expect(getByText(/fake_cta/).textContent).toEqual('fake_cta');
  expect(getByText(/fake_child/).textContent).toEqual('fake_child');
});