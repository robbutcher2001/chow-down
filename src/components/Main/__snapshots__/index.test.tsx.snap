// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main basic snapshot render 1`] = `
<main
  className="sc-AxhUy sc-AxgMl hQKHtI"
>
  <div>
    <section>
      <h2>
        fake_title
      </h2>
      <a
        className="sc-AxhCb iGINaf"
        href="/fake_link"
        onClick={[Function]}
      >
        fake_cta
      </a>
    </section>
    <div>
      fake_child
    </div>
  </div>
</main>
`;
